### PR TITLE
Fix Python and Shell sanity check warnings

### PR DESCRIPTION
**Pull request by jonathanhaigh-arm**
Format Python files using Black, and:

populate_rootfs_qca.sh:
* Tabs -> spaces.
* "command; if [ $? -ne 0 ]" -> "if ! command"

init.d.optee:
* Tabs -> spaces.
* "[ A -a B]" -> "[ A ] && [ B ]".
* "cat A | B" -> "B < A".

mbed-wrapper:
* Use "-r" with "read".

initramfs-init-script.sh:
* Use quotes.
* "cat A | B" -> "B < A".

mbl-hostname.sh:
* "[[ A && B ]]" -> "[ A ] && [ B ]"

create-partition-table-markdown:
* Modifying sys.path is required before imports here so ignore the
  warning that imports aren't at top of file.

create-update-payload:
* Modifying sys.path is required before imports here so ignore the
  warning that imports aren't at top of file.
* Don't exceed 79 chars per line.

non-rootfs-subdir.py:
rootfs-subdir.py:
signctl.py
* Docstrings and line widths.

setup-environment-internal:
* "command; if [ $? -eq 0 ]" -> "if command".
* "command; if [ $? -ne 0 ]" -> "if ! command"
* Use quotes.
* "$ARRAY" -> "${ARRAY[0]}" (Same meaning; clearer intent).
* "egrep" -> "grep -E"
* "[ A -a B]" -> "[ A ] && [ B ]".
* Shellcheck complained about:
    sha512sum "${OEROOT}"/setup-environment-internal 2>&1 > conf/checksum
  because the "2>&1" comes before the "> conf/checksum". I agree with
  shellcheck that it looks fishy, at least without a comment. I suspect
  the author wanted to send both sha512sum's stdout and stderr to the
  file, but I don't know why we'd want to send the stderr to the
  file, I don't know why we'd want to redirect the stderr at all, so
  just remove the stderr redirection.

**Jenkins links**
* http://jenkins.mbed-linux.arm.com/job/jij-warrior-dev/87/

**Lava links**
* http://e120856-lin.cambridge.arm.com/results.php?image_version=custom&image_number=87&custom_version=jij-warrior-dev&lava_query=&lava_owner=&failures=false&submitter=jjohnson&sort=by_device. Failures are:
  * Rootfs update failures. These are currently expected.
  * imx7d Wi-Fi test failure: https://lava.mbedcloudtesting.com/scheduler/job/73039. Looks like a problem during the board recovery procedure. @petedyerarm writes below that we currently get intermittent failures with recovery mode so I'm going to ignore this failure.

### CODE REVIEW

**Comment by jonathanhaigh-arm** [comment]
Thanks Rob.

The remaining warnings are:
```
/work/setup-env/setup-environment:28:1: note: Not following: setup-environment-internal was not specified as input (see shellcheck -x). [SC1091]
/work/setup-env/setup-environment-internal:192:1: warning: Use 'cd ... || exit' or 'cd ... || return' in case cd fails. [SC2164]
/work/setup-env/setup-environment-internal:271:8: warning: Declare and assign separately to avoid masking return values. [SC2155]
/work/setup-env/setup-environment-internal:274:8: warning: Declare and assign separately to avoid masking return values. [SC2155]
/work/setup-env/setup-environment-internal:336:32: warning: Use 'cd ... || exit' or 'cd ... || return' in case cd fails. [SC2164]
/work/meta-mbl-distro/recipes-core/mbl-initramfs-init/files/initramfs-init-script.sh:74:1: warning: Use 'cd ... || exit' or 'cd ... || return' in case cd fails. [SC2164]
```

When we discussed this IRL earlier today, I forgot to mention the initramfs-init-script.sh warning. It's in the same category as the setup-environment-internal warnings: the code currently doesn't do error checking properly anywhere in these scripts and fixing them will require more effort.

**Comment by diegorusso** [comment]
> Kicked off jenkins test: http://jenkins.mbed-linux.arm.com/blue/organizations/jenkins/jij-warrior-dev/detail/jij-warrior-dev/85/pipeline

There is still one error left:
```
/work/meta-mbl-distro/recipes-core/mbl-initramfs-init/files/initramfs-init-script.sh:74:1: warning: Use 'cd ... || exit' or 'cd ... || return' in case cd fails. [SC2164]
```

**Comment by jjohnson-arm** [comment]
> > Kicked off jenkins test: http://jenkins.mbed-linux.arm.com/blue/organizations/jenkins/jij-warrior-dev/detail/jij-warrior-dev/85/pipeline
> 
> There is still one error left:
> 
> ```
> /work/meta-mbl-distro/recipes-core/mbl-initramfs-init/files/initramfs-init-script.sh:74:1: warning: Use 'cd ... || exit' or 'cd ... || return' in case cd fails. [SC2164]
> ```
We know, but might as well test what we have overnight...

**Comment by jjohnson-arm** [comment]
http://jenkins.mbed-linux.arm.com/blue/organizations/jenkins/jij-warrior-dev/detail/jij-warrior-dev/86/pipeline

File: setup-env/setup-environment [hunk 1]
```diff
@@ -27,3 +27,6 @@
 
 # shellcheck disable=SC1091
 . setup-environment-internal
+
+# Clear the return trap
+trap - RETURN
```
[the remark this answers is not in the document]
Comment (jonathanhaigh-arm) [reply setup-env/setup-environment]: Oh, wait - there's already a shebang line but it says `/bin/sh` which is wrong. Fix coming...

**Comment by jonathanhaigh-arm** [comment]
New Jenkins build after all fixes: http://jenkins.mbed-linux.arm.com/job/jij-warrior-dev/87/

**Comment by petedyerarm** [comment]
We are seeing some intermittent issues with recovery mode where the device is either not connected or the bmap fails. We are investigating this infrastructure issue.

**Comment by jonathanhaigh-arm** [comment]
Thanks Pete.

**Comment by jonathanhaigh-arm** [comment]
I'veinteractively rebased this branch on warrior-dev so that I can split this PR into two commits. One containing the small fixes, and another for the setup-environment rewrite. Here's what I told `git` to do:
```
pick 5ba9a3b Format Python code using black
squash 22e09f3 Fix Python and Shell sanity check warnings
squash e1d681f fix signctl sanity checks
squash 09c88e6 Fix up initramfs to kernel panic on errors after watchdog setup
pick d118235 Rewrite setup environment scripts
squash 8f0ab9e Clear trap on setup env exit
squash e7f77fc setup-environment shebang -> bash
```

**Pull request merged**